### PR TITLE
feat: add pog log steps and pod label assertion step

### PR DESCRIPTION
**Pull request by danielhelfand**
Signed-off-by: Daniel Helfand <helfand.4@gmail.com>

### CODE REVIEW

File: pkg/kubernetes/kube.go [hunk 1]
```diff
@@ -725,6 +749,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+// TODO: Move this when other prs are merged in
+func isRetriable(err error) bool {
+	for _, msg := range retriableErrors {
+		if strings.Contains(err.Error(), msg) {
+			return true
+		}
+	}
+	return false
+}
+
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go]: already exist in common.go and util.go

File: pkg/kubernetes/kube.go [hunk 2]
```diff
@@ -725,6 +749,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+func listPodsWithLabelSelector(client kubernetes.Interface, namespace, selector string) (*corev1.PodList, error) {
+	pods, err := retryOnError(&DefaultRetry, isRetriable, func() (interface{}, error) {
+		return client.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: selector})
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list pods")
+	}
+
+	return pods.(*corev1.PodList), nil
+}
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go]: Already exists in kube_helpers.go

File: pkg/kubernetes/kube.go [hunk 3]
```diff
@@ -725,6 +749,238 @@ func (kc *Client) ClusterRbacIsFound(resource, name string) error {
...
+	// for gracefully exiting failures.
+	// https://github.com/onsi/ginkgo/blob/v1.16.5/ginkgo_dsl.go#L283-L303
+	gomega.RegisterFailHandler(ginkgo.Fail)
+}
+
+func (kc *Client) ThePodsInNamespaceWithSelectorHasThisSentenceInLogsSinceTime(namespace, selector, searchkeyword, sinceTime string, timeout int) error {
+	gomega.Eventually(func() error {
+		if kc.KubeInterface == nil {
+			return errors.Errorf("'Client.KubeInterface' is nil. 'AKubernetesCluster' sets this interface, try calling it before using this method")
+		}
```
Comment (Kyle-Wong) [inline pkg/kubernetes/kube.go:762]: Note to replace instances of this with kc.Validate() once Alfredo's PR merges

File: kubedog.go [hunk 4]
```diff
@@ -69,6 +69,12 @@ func (kdt *Test) Run() {
...
 	kdt.scenarioContext.Step(`^(?:I )?(add|remove) cluster shared iam role$`, kdt.AwsContext.ClusterSharedIamOperation)
+  kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) has ([^"]*) in logs since ([^"]*) time and times out in (\S+) seconds`, kdt.KubeContext.ThePodsInNamespaceWithSelectorHasThisSentenceInLogsSinceTime)
+	kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) doesn't have ([^"]*) in logs since ([^"]*) time$`, kdt.KubeContext.NoMatchingStringInLogsSinceTime)
+	kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) have no errors in logs since ([^"]*) time$`, kdt.KubeContext.ThePodsInNamespaceWithSelectorHaveNoErrorsInLogsSinceTime)
+	kdt.scenarioContext.Step(`^the pods in namespace ([^"]*) with selector ([^"]*) have some errors in logs since ([^"]*) time$`, kdt.KubeContext.ThePodsInNamespaceWithSelectorHaveSomeErrorsInLogsSinceTime)
+	kdt.scenarioContext.Step(`^the pod ([^"]*) in namespace ([^"]*) should have labels ([^"]*)$`, kdt.KubeContext.ThePodInNamespaceShouldHaveLabels)
```
Comment (Kyle-Wong) [inline kubedog.go]: Make initial `the` optional

Comment (Kyle-Wong) [inline kubedog.go:76]: Does this need to be specifically 1 pod or can it be changed to `pods` for consistency with the above steps?

Comment (danielhelfand) [reply kubedog.go]: Changed to use selector instead of pod name. Now supports multiple pods.

Comment (danielhelfand) [reply kubedog.go]: Done

Comment (Kyle-Wong) [reply kubedog.go]: Swap the order of namespace and selector parameters in step syntax for consistency

Comment (Kyle-Wong) [reply kubedog.go]: “the” is still not optional, it should be like “(:the )?”

Comment (danielhelfand) [reply kubedog.go]: My misunderstanding. I thought `^` accomplished this, but that's my mistake. Are you suggesting to make the optional for all steps beginning with the, or just the step on line 76?

Comment (danielhelfand) [reply kubedog.go]: Looking at other steps, probably all of them. I can change.

Comment (danielhelfand) [reply kubedog.go]: Done

Comment (Kyle-Wong) [reply kubedog.go]: Yep, all of the steps. It uses standard regex btw if you want to understand it

Comment (danielhelfand) [reply kubedog.go]: Done

**Review by Kyle-Wong — APPROVED** [review]

**Pull request closed without merging**